src/libstd/panicking.rs

```diff
@@ -322,19 +324,36 @@ pub unsafe fn r#try<R, F: FnOnce() -> R>(f: F) -> Result<R, Box<dyn Any + Send>>
     let mut data = Data { f: ManuallyDrop::new(f) };
 
     let data_ptr = &mut data as *mut _ as *mut u8;
-    return if intrinsics::r#try(do_call::<F, R>, data_ptr, do_catch::<F, R>) == 0 {
-        Ok(ManuallyDrop::into_inner(data.r))
-    } else {
-        Err(ManuallyDrop::into_inner(data.p))
-    };
+    // SAFETY:
+    //
+    // Access to the union's fields: they are made safe by:
+    //  - 1) The call to `into_inner` will allow the data to be dropped later,
+    //    avoiding leaks.
```

Comment:
This does not belong in the safety comment; leaking memory is not a safety issue. The safety comment should solely be about why this is safe, in this case it's essentially "This is std and std knows that the try intrinsic fills in the `r` or `p` union field based on its return value".